lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -2078,12 +2090,55 @@ void GrandCentralPass::runOnOperation() {
     VerbatimBuilder::Base verbatimData;
     VerbatimBuilder verbatim(verbatimData);
     verbatim += instanceSymbol;
-    auto iface =
-        traverseBundle(bundle, bundle.getID(), bundle.getPrefix(), verbatim);
+    // List of interface elements.
+    SmallVector<TypeSum> interfaceElems;
+    // insertedOps will record all the verbatim ops inserted in the companion
+    // module, which must be erased, if the interface is redundant. The
+    // traverseBundle is a recursive traversal of the AugmentedBundleType, it
+    // will construct the Interface and populate it with the elements and insert
+    // the corresponding verbatim xmr assignments. The interface might be
+    // redundant if it is a nonlocal view and another exactly same interface is
+    // already generated. To compare the interface elements, with an existing
+    // interface, it is necessary to complete the traverseBundle traversal and
+    // gather all the elements of the interface.
+    insertedOps.clear();
+    auto iface = traverseBundle(bundle, bundle.getID(), bundle.getPrefix(),
+                                verbatim, interfaceElems);
     if (!iface) {
       removalError = true;
       continue;
     }
+
+    // If the companion module has two exactly same ViewAnnotation.companion
+    // annotations, then add the interface for only one of them. This happens
+    // when the companion is deduped.
+    auto viewMapIter = companionToInterfaceMap.find(companionModule);
+    if (viewMapIter != companionToInterfaceMap.end()) {
+      bool sameInterface = true;
+      if (viewMapIter->getSecond().size() != interfaceElems.size())
+        sameInterface = false;
+      else
+        // Now compare  the existing interface with this.
+        for (auto &&[prevElem, thisElem] :
+             llvm::zip(viewMapIter->getSecond(), interfaceElems)) {
+          if ((prevElem.index() == thisElem.index()) &&
+              ((prevElem.index() == 1 && *std::get_if<Type>(&prevElem) ==
+                                             *std::get_if<Type>(&thisElem)) ||
+               (prevElem.index() == 0 &&
+                std::get_if<VerbatimType>(&prevElem) ==
+                    std::get_if<VerbatimType>(&thisElem))))
```

Comment:
How does the `index()` control whether or not this is a logic or a sub-interface?

Reply:
This was not handling sub-interfaces correctly. Updated the implementation to just record the interface signal ops.